controller/src/main/java/org/apache/rocketmq/controller/impl/JRaftController.java

```diff
@@ -0,0 +1,280 @@
+import org.apache.rocketmq.remoting.protocol.body.SyncStateSet;
+import org.apache.rocketmq.remoting.protocol.header.controller.AlterSyncStateSetRequestHeader;
+import org.apache.rocketmq.remoting.protocol.header.controller.ElectMasterRequestHeader;
+import org.apache.rocketmq.remoting.protocol.header.controller.GetMetaDataResponseHeader;
+import org.apache.rocketmq.remoting.protocol.header.controller.GetReplicaInfoRequestHeader;
+import org.apache.rocketmq.remoting.protocol.header.controller.admin.CleanControllerBrokerDataRequestHeader;
+import org.apache.rocketmq.remoting.protocol.header.controller.register.ApplyBrokerIdRequestHeader;
+import org.apache.rocketmq.remoting.protocol.header.controller.register.GetNextBrokerIdRequestHeader;
+import org.apache.rocketmq.remoting.protocol.header.controller.register.RegisterBrokerToControllerRequestHeader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class JRaftController implements Controller {
+    private static final Logger log = LoggerFactory.getLogger(LoggerName.CONTROLLER_LOGGER_NAME);
+    private final RaftGroupService raftGroupService;
+    private final Node node;
+    private final JRaftControllerStateMachine stateMachine;
+    private final ControllerConfig controllerConfig;
+    private final List<BrokerLifecycleListener> brokerLifecycleListeners;
+    private final Map<PeerId/* jRaft peerId */, String/* Controller RPC Server Addr */> peerIdToAddr;
+    private final NettyRemotingServer remotingServer;
+
+    public JRaftController(ControllerConfig controllerConfig,
+        final ChannelEventListener channelEventListener) throws IOException {
+        this.controllerConfig = controllerConfig;
+        this.brokerLifecycleListeners = new ArrayList<>();
+
+        final NodeOptions nodeOptions = new NodeOptions();
+        nodeOptions.setElectionTimeoutMs(controllerConfig.getjRaftElectionTimeoutMs());
+        nodeOptions.setSnapshotIntervalSecs(controllerConfig.getjRaftSnapshotIntervalSecs());
+        final PeerId serverId = new PeerId();
+        if (!serverId.parse(controllerConfig.getjRaftServerId())) {
+            throw new IllegalArgumentException("Fail to parse serverId:" + controllerConfig.getjRaftServerId());
+        }
+        final Configuration initConf = new Configuration();
+        if (!initConf.parse(controllerConfig.getjRaftInitConf())) {
+            throw new IllegalArgumentException("Fail to parse initConf:" + controllerConfig.getjRaftInitConf());
+        }
+        nodeOptions.setInitialConf(initConf);
+
+        FileUtils.forceMkdir(new File(controllerConfig.getControllerStorePath()));
+        nodeOptions.setLogUri(controllerConfig.getControllerStorePath() + File.separator + "log");
```

Comment:
It seems that the path will always be "xxx/DLedgerController/xxxx", even if i set the controllerType as "jRaft".